edgePy/io/_io.py

```diff
@@ -7,15 +7,60 @@
 
 
 __all__ = [
-    'Importer',
-    'get_dataset_path'
+    'GroupImporter',
+    'DataImporter',
+    'get_dataset_path',
+    'get_open_function'
 ]
 
 
-class Importer(object):
+def get_open_function(filename):
+    decode_required = False
+    if filename.endswith("gz"):
+        open_function = gzip.open
+        decode_required = True
+    else:
+        open_function = open
+
+    return open_function, decode_required
+
+
+class GroupImporter(object):
+
+    def __init__(self, filename=None):
+        self.filename = str(filename) if filename else None
+        if not self.filename:
+            raise Exception("No group filename")
+        self.samples = {}
+        self.groups = {}
+        self.read_file()
+
+    def read_file(self):
+        open_function, decode_required = get_open_function(filename=self.filename)
+        with open_function(self.filename) as f:
+            for line in f:
+                if decode_required:
+                    # this is needed if we are using gzip, which returns a
+                    # binary-string.
+                    line = line.decode('utf-8')
+                line = line.strip().split(":")
+                if len(line) < 2:
+                    continue
+                group = line[0].strip()
+                samples = [x.strip() for x in line[1].split(",")]
+                self.groups[group] = samples
+                for sample in samples:
+                    if sample in self.samples:
+                        raise Exception(f"Duplicate sample detected! {sample}")
+                    self.samples[sample] = group
```

Comment:
Confused on what the two dictionaries are for. `self.samples` / `self.groups`. Are the reverse mappings of each other?

Docs needed (can put in docstring).